libs/extensions/std/exec/src/archive-interpreter-executor.spec.ts

```diff
@@ -0,0 +1,160 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import * as path from 'path';
+
+import * as R from '@jvalue/jayvee-execution';
+import { getTestExecutionContext } from '@jvalue/jayvee-execution/test';
+import { StdLangExtension } from '@jvalue/jayvee-extensions/std/lang';
+import {
+  BlockDefinition,
+  IOType,
+  createJayveeServices,
+  useExtension,
+} from '@jvalue/jayvee-language-server';
+import {
+  ParseHelperOptions,
+  TestLangExtension,
+  expectNoParserAndLexerErrors,
+  parseHelper,
+  readJvTestAssetHelper,
+} from '@jvalue/jayvee-language-server/test';
+import { AstNode, AstNodeLocator, LangiumDocument } from 'langium';
+import { NodeFileSystem } from 'langium/node';
+
+import { createBinaryFileFromLocalFile } from '../test';
+
+import { ArchiveInterpreterExecutor } from './archive-interpreter-executor';
+
+describe('Validation of ArchiveInterpreterExecutor', () => {
+  let parse: (
+    input: string,
+    options?: ParseHelperOptions,
+  ) => Promise<LangiumDocument<AstNode>>;
+
+  let locator: AstNodeLocator;
+
+  const readJvTestAsset = readJvTestAssetHelper(
+    __dirname,
+    '../test/assets/archive-interpreter-executor/',
+  );
+
+  function readTestArchive(fileName: string): R.BinaryFile {
+    const absoluteFileName = path.resolve(
+      __dirname,
+      '../test/assets/archive-interpreter-executor/',
+      fileName,
+    );
+    return createBinaryFileFromLocalFile(absoluteFileName);
+  }
+
+  async function parseAndExecuteArchiveInterpreter(
+    input: string,
+    IOInput: R.BinaryFile,
+  ): Promise<R.Result<R.FileSystem>> {
+    const document = await parse(input, { validationChecks: 'all' });
+    expectNoParserAndLexerErrors(document);
+
+    const block = locator.getAstNode<BlockDefinition>(
+      document.parseResult.value,
+      'pipelines@0/blocks@1',
```

Comment:
I could do that, but in my opinion there should not be multiple blocks to test in a single jv file.
Also that would mean most tests would fail, because the test would have to handle all blocks.

For example:
- jv file with two `ArchiveInterpreter` blocks to test
- One block for testing zip archives and one for gz archives
- Test for zip block would also execute gz block and automatically fail because input is a zip file

Reply:
Discussed in person